core/vmpu/src/vmpu.c

```diff
@@ -248,6 +248,18 @@ static void vmpu_sanity_checks_box_cfgtbl(uint8_t box_id, UvisorBoxConfig const
                    box_id, (uint32_t) box_cfgtbl, box_cfgtbl->stack_size, UVISOR_MIN_STACK_SIZE);
     }
 
+    /* Checks specific to the public box */
+    if (box_id == 0) {
+        if (box_cfgtbl->stack_size != 0) {
+            HALT_ERROR(SANITY_CHECK_FAILED, "Box %i @0x%08X: The public box stack size must be set to 0 (found %uB).\r\n",
+                       box_id, (uint32_t) box_cfgtbl, box_cfgtbl->stack_size);
+        }
+        if (box_cfgtbl->bss.size_of.heap != 0) {
+            HALT_ERROR(SANITY_CHECK_FAILED, "Box %i @0x%08X: The public box heap size must be set to 0 (found %uB).\r\n",
+                       box_id, (uint32_t) box_cfgtbl, box_cfgtbl->bss.size_of.heap);
+        }
+    }
```

Comment:
Added a comment that should clarify it. See also the related comments in `vmpu_box_index_init`:

```C
        /* FIXME: At the moment we cannot just reliably use the size of the heap
         *        as provided by the configuration table, because the public box
         *        has a variable heap size that is not identified at compile
         *        time. We should remove this dependency. */
```

```C
        /* Set the heap pointer for the public box.
         * This value must be set manually because the previous loop will have
         * caught a zero heap size and thus set the pointer to NULL. */
```